app/components/input.tsx

```diff
@@ -1,45 +1,46 @@
-import { FC, ChangeEventHandler  } from "react";
+import clsx from "clsx";
 
-interface Props {
-  labelText: string;
-  placeholder: string;
-  type: string;
-  required: boolean;
-  name?: string;
-  disabled?: boolean;
-  value?: string | number;
-  onChange?: ChangeEventHandler<HTMLInputElement>;
-}
+type Props = React.InputHTMLAttributes<HTMLInputElement> & {
+  labelText?: string;
+};
+
+export function Input({ labelText, ...props }: Props) {
+  const baseClasses = [
+    "form-control",
+    "block",
+    "w-full",
+    "px-4",
+    "py-2",
+    "text-base",
+    "bg-white",
+    "bg-clip-padding",
+    "border",
+    "border-solid",
+    "rounded-full",
+  ];
+
+  const comlementaryClasses = !props.disabled
+    ? [
+        "font-normal",
+        "text-gray-600",
+        "border-[#FFC905]",
+        "hover:shadow-lg",
+        "transition",
+        "ease-in-out",
+        "m-0",
+        "focus:text-black",
+        "focus:bg-white",
+        "focus:border-[#FF8C05]",
+        "focus:outline-none",
+      ]
+    : ["font-light", "text-gray-200", "border-[#FFF2CE] "];
```

Comment:
`disabled` prop is not working.

The input keeps the same style even if it's disabled.